### PR TITLE
fix(floating-ui): improve floating element performance

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #7979 #8214 #8386 #8419 #5697

## Summary

- reverts #8001
  - Keeps doc updates 
- reverts partially #8230
  - Keeps drag classes that are setting `overflow:hidden` when a drag is occurring.
  - Keeps roundByDPR floating-ui fix. https://floating-ui.com/docs/misc#subpixel-and-accelerated-positioning
- Hides overflow for `calcite-sortable--chosen` class
- Only sets `top` and `left` on floating element once positioned. **This fixes native drag and drop issue**

### CODE REVIEW

File: packages/calcite-components/src/utils/floating-ui.ts [hunk 1]
```diff
@@ -489,8 +494,6 @@ export function connectFloatingUI(
 
     // initial positioning based on https://floating-ui.com/docs/computePosition#initial-layout
     position: component.overlayPositioning,
-    top: "0",
```
Comment (driskull) [inline packages/calcite-components/src/utils/floating-ui.ts:492]: @jcfranco do you know why this was here?

It was added here: https://github.com/Esri/calcite-design-system/commit/e220686fa366b86dda2647da25144eb45a4a193d#diff-3c419c81de1cc684c4d1d859881782bc87d089da410675b5af3f6c468b95ef13R458

I'd like to remove this to fix the drag and drop positioning issue. #7979

Comment (driskull) [reply packages/calcite-components/src/utils/floating-ui.ts:492]: Instead, this is only set once an actual position occurs.

**Review by jcfranco — APPROVED** [review]
🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎
🏎🎈🎈🎈🎈🏎🎈🏎🏎🏎🏎🏎🎈🏎🎈🎈🎈🎈🏎🏎🎈🎈🎈🏎🏎🎈🎈🏎🏎🎈🏎🏎🏎🎈🏎🎈🎈🎈🎈🏎
🏎🎈🏎🏎🎈🏎🎈🏎🏎🏎🏎🏎🎈🏎🎈🏎🏎🏎🏎🎈🏎🏎🏎🏎🎈🏎🏎🎈🏎🎈🎈🏎🎈🎈🏎🎈🏎🏎🏎🏎
🏎🎈🎈🎈🎈🏎🎈🏎🏎🎈🏎🏎🎈🏎🎈🎈🎈🏎🏎🏎🎈🎈🏎🏎🎈🏎🏎🎈🏎🎈🏎🎈🏎🎈🏎🎈🎈🎈🏎🏎
🏎🎈🏎🏎🎈🏎🎈🏎🎈🏎🎈🏎🎈🏎🎈🏎🏎🏎🏎🏎🏎🏎🎈🏎🎈🏎🏎🎈🏎🎈🏎🏎🏎🎈🏎🎈🏎🏎🏎🏎
🏎🎈🏎🏎🎈🏎🏎🎈🏎🏎🏎🎈🏎🏎🎈🎈🎈🎈🏎🎈🎈🎈🏎🏎🏎🎈🎈🏎🏎🎈🏎🏎🏎🎈🏎🎈🎈🎈🎈🏎
🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎🏎

Do we have a perf impact issue we could link to for this PR?

**Comment by driskull** [comment]
> Do we have a perf impact issue we could link to for this PR?

No, can we get one? cc @paulcpederson

File: packages/calcite-components/src/components/dropdown/dropdown.tsx [hunk 2]
```diff
@@ -78,12 +78,14 @@ export class Dropdown
...
-      onToggleOpenCloseComponent(this);
+    onToggleOpenCloseComponent(this);
+
+    if (this.disabled) {
+      this.open = false;
       return;
     }
 
-    this.open = false;
+    this.reposition(true);
```
Comment (jcfranco) [inline packages/calcite-components/src/components/dropdown/dropdown.tsx:88]: Great catch on this perf fix, btw!

File: packages/calcite-components/src/utils/floating-ui.ts (hunk 1 above)
Comment (driskull) [reply packages/calcite-components/src/utils/floating-ui.ts:492]: Looks like that doc says only the `position` needs to be there initially so the top and left can happen later.

**Comment by jcfranco** [comment]
@driskull Could you also check if this introduces a regression for https://github.com/Esri/calcite-design-system/issues/5697? 🙇‍♂️

**Comment by paulcpederson** [comment]
https://github.com/Esri/calcite-design-system/issues/8419

**Comment by driskull** [comment]
>@driskull Could you also check if this introduces a regression for https://github.com/Esri/calcite-design-system/issues/5697? 🙇‍♂️

@jcfranco It seems like this would be a regression here. This makes sense because the component doesn't get its position necessary for a proper animation until after its opened. Once it has been opened once the position is there and the correct animation takes place.

I think we'll have to accept this for now. Thoughts?

**Comment by jcfranco** [comment]
> I think we'll have to accept this for now. Thoughts?

Agreed. Let's reopen the issue and comment on the reasoning. cc @geospatialem

**Pull request merged**